src/shared/copy.c

```diff
@@ -1594,3 +1596,49 @@ int copy_xattr(int df, const char *from, int dt, const char *to, CopyFlags copy_
 
         return ret;
 }
+
+int reflink(int infd, int outfd) {
+        int r;
+
+        assert(infd >= 0);
+        assert(outfd >= 0);
+
+        /* Make sure we invoke the ioctl on a regular file, so that no device driver accidentally gets it. */
+
+        r = fd_verify_regular(outfd);
+        if (r < 0)
+                return r;
+
+        /* FICLONE was introduced in Linux 4.5, so let's fall back to BTRFS_IOC_CLONE if it's not supported. */
+
+        r = ioctl(outfd, FICLONE, infd);
```

Comment:
The doc says we support 4.3, do we need an ifndef -> define for this and the other one?